cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/deploy/RemoteProgramRunDispatcher.java

```diff
@@ -118,13 +115,18 @@ public ProgramController dispatchProgram(ProgramRunDispatcherInfo programRunDisp
         programId.getType());
     RunRecordDetail runRecordDetail = store.getRun(programRunId);
     TwillController twillController = null;
-    if (remoteTwillRunnerService != null && runRecordDetail != null) {
+    if (runRecordDetail != null) {
       twillController = remoteTwillRunnerService.createTwillControllerFromRunRecord(runRecordDetail);
     }
     ProgramController programController = null;
-    if (runner instanceof DistributedWorkflowProgramRunner && twillController != null) {
-      programController =
-        ((DistributedWorkflowProgramRunner) runner).createProgramController(programRunId, twillController);
+    if (runner instanceof ProgramControllerCreator && twillController != null) {
+      programController = ((ProgramControllerCreator) runner).createProgramController(programRunId, twillController);
+    }
+    if (Objects.isNull(programController)) {
```

Comment:
no need to do this, just check for null with `== null`. Objects.isNull is mainly useful when you want to use it as a method reference as `Objects::isNull`.